src/pages/EnablePayments/EnablePaymentsPage.js

```diff
@@ -54,31 +53,37 @@ class EnablePaymentsPage extends React.Component {
             return <FullScreenLoadingIndicator />;
         }
 
-        if (this.props.userWallet.errorCode === CONST.WALLET.ERROR.KYC) {
-            return (
-                <ScreenWrapper style={[styles.flex1]} keyboardAvoidingViewBehavior="height">
-                    <HeaderWithCloseButton
-                        title={this.props.translate('additionalDetailsStep.headerTitle')}
-                        onCloseButtonPress={() => Navigation.dismissModal()}
-                    />
-                    <FailedKYC />
-                </ScreenWrapper>
-            );
-        }
-        if (this.props.userWallet.shouldShowWalletActivationSuccess) {
-            return (
-                <ActivateStep userWallet={this.props.userWallet} />
-            );
-        }
-
-        const currentStep = this.props.userWallet.currentStep || CONST.WALLET.STEP.ADDITIONAL_DETAILS;
-
         return (
             <ScreenWrapper>
-                {currentStep === CONST.WALLET.STEP.ADDITIONAL_DETAILS && <AdditionalDetailsStep walletAdditionalDetailsDraft={this.props.walletAdditionalDetailsDraft} />}
-                {currentStep === CONST.WALLET.STEP.ONFIDO && <OnfidoStep walletAdditionalDetailsDraft={this.props.walletAdditionalDetailsDraft} />}
-                {currentStep === CONST.WALLET.STEP.TERMS && <TermsStep />}
-                {currentStep === CONST.WALLET.STEP.ACTIVATE && <ActivateStep userWallet={this.props.userWallet} />}
+                {(() => {
```

Comment:
IIFE is looking very awkward here.

Reply:
You don't like? I would review another refactor if you feel strongly – I thought this was pretty readable and like having only one parent `<ScreenWrapper>`

Reply:
Yeah, combined statements are good but IIFE in JSX is not very common.

Reply:
Came across this randomly, but yeah I think I don't like either. Feels like an anti-pattern and an alternate approach would be to create a new component for all the logic like:

```jsx
<ScreenWrapper>
    <EnablePaymentsContent {...props} />
</ScreenWrapper>
```

Reply:
I'm not opposed and that sounds like a good alternative @marcaaron. Feel free to create an issue if you want 👍🏼